juju/juju
# Revert deletion of the Oracle Classic provider

This commit reverts the deletion of the Oracle Classic provider, while keeping the changes to the feature tests made to enable its removal in-place. 

Note: OCI is the only provider available via `juju list-clouds`, but the "oracle" cloud type is available for any pre-existing cloud definitions. In essence, this retains backwards compatibility without advertising oracle classic's availability.  

## QA steps

Ensure that Oracle Classic clouds can still be operated on via Juju.

Here is the current output from `juju list-clouds`:

```plain
$ juju list-clouds
Cloud           Regions  Default          Type        Description
aws                  15  us-east-1        ec2         Amazon Web Services
aws-china             2  cn-north-1       ec2         Amazon China
aws-gov               1  us-gov-west-1    ec2         Amazon (USA Government)
azure                27  centralus        azure       Microsoft Azure
azure-china           2  chinaeast        azure       Microsoft Azure China
cloudsigma           12  dub              cloudsigma  CloudSigma Cloud
google               18  us-east1         gce         Google Cloud Platform
joyent                6  us-east-1        joyent      Joyent Cloud
oracle                4  us-phoenix-1     oci         Oracle Cloud Infrastructure
oracle-classic        5  uscom-central-1  oracle      Oracle Cloud Infrastructure Classic
rackspace             6  dfw              rackspace   Rackspace Cloud
localhost             1  localhost        lxd         LXD Container Hypervisor
```

## Documentation changes

n/a

## Bug reference

n/a
